src/strands/tools/executor.py

```diff
@@ -74,14 +73,14 @@ def work(
     ) -> ToolResult:
         events = handle(tool)
 
-        while True:
-            try:
+        try:
+            while True:
                 event = next(events)
                 worker_queue.put((worker_id, event))
                 worker_event.wait()
 
-            except StopIteration as stop:
-                return cast(ToolResult, stop.value)
+        except StopIteration as stop:
```

Comment:
I see what you mean about the syntax on this being weird, but I do like the implementor side.

Could we use this for model_providers too so that `stop_reason` returns are more apparent?